test/jdk/java/awt/ScrollPane/ScrollPaneLimitation/ScrollPaneLimitation.java

```diff
@@ -0,0 +1,174 @@
+import java.awt.Component;
+import java.awt.Container;
+import java.awt.Dimension;
+import java.awt.EventQueue;
+import java.awt.Frame;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.ScrollPane;
+
+import java.awt.event.InputEvent;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseAdapter;
+
+import java.util.Properties;
+
+public class ScrollPaneLimitation {
+    final static int SCROLL_POS = 50000;
+    public static boolean mouseWasPressed = false;
+    public static Component child = null;
+    private Object lock = new Object();
+    public Frame frame;
+    Robot robot;
+    Point p;
+    ScrollPane pane;
+
+    public static void main(String[] args) throws Exception {
+        ScrollPaneLimitation scrollTest = new ScrollPaneLimitation();
+
+        scrollTest.init();
+        scrollTest.start();
+    }
+
+    public void init() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            frame = new Frame("Scroll Pane Limitation");
+            frame.setLayout(new BorderLayout());
+        });
+        robot = new Robot();
+    }
+
+    public void start() throws Exception {
+
+        // Check for Win32 environment. This test is only for Win32.
+        // Even more: not for Windows 98: it doesn't support scroll children that long.
+        try {
+            Properties prop = System.getProperties();
+
+            String os = prop.getProperty("os.name", "");
+            if (os != null &&
+                    (os.indexOf("98") == -1) &&
+                    (os.startsWith("Win") || os.startsWith("win") ||
+                            os.startsWith("WIN"))) {
+
+                EventQueue.invokeAndWait(() -> {
+                    pane = new ScrollPane();
+                    frame.add(pane);
+                    child = new MyPanel();
+                    child.addMouseListener(new MouseAdapter() {
+                        public void mousePressed(MouseEvent e) {
+                            if (e.getID() == MouseEvent.MOUSE_PRESSED) {
+                                if (e.getSource() == ScrollPaneLimitation.child
+                                        && e.getY() > SCROLL_POS) {
+                                    mouseWasPressed = true;
+                                    synchronized (lock) {
+                                        lock.notify();
+                                    }
+                                }
+                            }
+                        }
+                    });
+                    pane.add(child);
+                    frame.setSize(200, 200);
+                    frame.pack();
+                    frame.setLocationRelativeTo(null);
+                    frame.setVisible(true);
+                    pane.doLayout();
+
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException("test was interrupted");
+                    }
```

Comment:
```suggestion
                    robot.delay(1000);
```
Much cleaner, isn't it?

Reply:
Done.